src/sparse/KokkosSparse_gauss_seidel.hpp

```diff
@@ -49,611 +49,689 @@
+                     "KokkosSparse::gauss_seidel_symbolic: Size type of the matrix should be same as kernelHandle sizetype.");
+
+      static_assert (std::is_same<typename KernelHandle::const_nnz_lno_t,
+                     typename lno_nnz_view_t_::const_value_type>::value,
+                     "KokkosSparse::gauss_seidel_symbolic: lno type of the matrix should be same as kernelHandle lno_t.");
+
+      static_assert (std::is_same<typename KernelHandle::const_nnz_scalar_t,
+                     typename scalar_nnz_view_t_::const_value_type>::value,
+                     "KokkosSparse::gauss_seidel_symbolic: scalar type of the matrix should be same as kernelHandle scalar_t.");
+
+
+      typedef typename KernelHandle::const_size_type c_size_t;
+      typedef typename KernelHandle::const_nnz_lno_t c_lno_t;
+      typedef typename KernelHandle::const_nnz_scalar_t c_scalar_t;
+
+      typedef typename KernelHandle::HandleExecSpace c_exec_t;
+      typedef typename KernelHandle::HandleTempMemorySpace c_temp_t;
+      typedef typename KernelHandle::HandlePersistentMemorySpace c_persist_t;
+
+      typedef typename  KokkosKernels::Experimental::KokkosKernelsHandle<c_size_t, c_lno_t, c_scalar_t, c_exec_t, c_temp_t, c_persist_t> const_handle_type;
+      //const_handle_type tmp_handle = *handle;
+      const_handle_type tmp_handle (*handle);
+
+      typedef Kokkos::View<
+        typename lno_row_view_t_::const_value_type*,
+        typename KokkosKernels::Impl::GetUnifiedLayout<lno_row_view_t_>::array_layout,
+        typename lno_row_view_t_::device_type,
+        Kokkos::MemoryTraits<Kokkos::Unmanaged> > Internal_alno_row_view_t_;
+
+      typedef Kokkos::View<
+        typename lno_nnz_view_t_::const_value_type*,
+        typename KokkosKernels::Impl::GetUnifiedLayout<lno_nnz_view_t_>::array_layout,
+        typename lno_nnz_view_t_::device_type,
+        Kokkos::MemoryTraits<Kokkos::Unmanaged> > Internal_alno_nnz_view_t_;
+
+      typedef Kokkos::View<
+        typename scalar_nnz_view_t_::const_value_type*,
+        typename KokkosKernels::Impl::GetUnifiedLayout<scalar_nnz_view_t_>::array_layout,
+        typename scalar_nnz_view_t_::device_type,
+        Kokkos::MemoryTraits<Kokkos::Unmanaged> > Internal_ascalar_nnz_view_t_;
+
+      Internal_alno_row_view_t_ const_a_r (row_map.data(), row_map.extent(0));
+      Internal_alno_nnz_view_t_ const_a_l (entries.data(), entries.extent(0));
+      Internal_ascalar_nnz_view_t_ const_a_v (values.data(), values.extent(0));
+      Internal_ascalar_nnz_view_t_ const_a_d (given_inverse_diagonal.data(), given_inverse_diagonal.extent(0));
+
+
+      using namespace KokkosSparse::Impl;
+
+      GAUSS_SEIDEL_NUMERIC<const_handle_type, Internal_alno_row_view_t_, Internal_alno_nnz_view_t_, Internal_ascalar_nnz_view_t_>::gauss_seidel_numeric
+        (&tmp_handle, num_rows, num_cols, const_a_r, const_a_l, const_a_v, const_a_d, is_graph_symmetric);
```

Comment:
@srajama1 The diagonal is passed here as `const_a_d`.

Reply:
Thanks, missed the basic thing.